src/app/forms/components/switchable-input/switchable-input.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { FormNodeEditTypes, FormNodeOption, FormNodeWidth } from '@forms/services/dynamic-form.types';
+import { Observable, of } from 'rxjs';
+
+@Component({
+  selector: 'app-switchable-input[form][type][name][isEditing]',
+  templateUrl: './switchable-input.component.html',
+  styleUrls: ['./switchable-input.component.scss']
+})
+export class SwitchableInputComponent implements OnInit {
+  @Input() type!: FormNodeEditTypes;
+  @Input() form!: FormGroup;
+  @Input() name!: string;
+
+  @Input() isEditing = true;
+
+  @Input() label?: string;
+  @Input() suffix?: string;
+  @Input() width?: FormNodeWidth;
+  @Input() options?: Array<FormNodeOption<string | number | boolean>>;
+
+  demilimitor = { regex: '\\. (?<!\\..\\. )', separator: '. ' };
+
+  ngOnInit(): void {
+    if (this.requiresOptions && !this.options) {
+      throw new Error('Cannot use autocomplete or radio control without providing an options array.');
+    }
+  }
+
+  get requiresOptions(): boolean {
+    return this.type === this.types.AUTOCOMPLETE
+      || this.type === this.types.CHECKBOX
+      || this.type === this.types.DROPDOWN
+      || this.type === this.types.RADIO;
+  }
+
+  get options$(): Observable<Array<FormNodeOption<string | number | boolean>>> {
```

Comment:
there should be an interface for this type
```suggestion
  get options$(): Observable<MultiOptions> {
```

Reply:
Done